Trove.js

```diff
@@ -0,0 +1,334 @@
+					url: item.url,
+					mimeType: 'text/html',
+					snapshot: false
+				})
+			}
+
+			// This gives a better version-aware url.
+			item.url = ZU.xpathText(doc, "//meta[@property='og:url']/@content");
+			item.abstractNote = ZU.xpathText(doc, "//meta[@property='og:description']/@content");
+			delete item.itemID;
+			delete item.type;
+
+			// Add tags
+			tags = ZU.xpath(doc, "//div[@id='tagswork']/ul/li");
+			for (var i = 0; i < tags.length; i++) {
+				tag = ZU.xpathText(tags[i], "a");
+				item.tags.push(tag);
+			}
+
+			if (thumbnailURL !== null) {
+				item.attachments.push({
+					url: thumbnailURL, 
+					title: 'Trove thumbnail image', 
+					mimeType:'image/jpeg'
+				});
+			}
+			item.complete();
+		});
+		translator.translate();
+	});
+}
+
+/** BEGIN TEST CASES **/
+var testCases = [
+	{
+		"type": "web",
+		"url": "http://trove.nla.gov.au/version/11567057",
+		"items": [
+			{
+				"itemType": "book",
+				"title": "Experiences of a meteorologist in South Australia",
+				"creators": [
+					{
+						"firstName": "Clement L. (Clement Lindley)",
+						"lastName": "Wragge",
+						"creatorType": "author"
+					}
+				],
+				"date": "1980",
+				"ISBN": "9780908065073",
+				"abstractNote": "In 14 libraries. 24 p. : ill. ; 22 cm. Wragge, Clement L. (Clement Lindley), 1852-1922. South Australia. Climate, 1883-1884. Meteorologists -- South Australia -- Biography. South Australia -- Climate -- History.",
+				"extra": "Reprinted from Good words for 1887/ edited by Donald Macleod, published: London: Isbister and Co",
+				"language": "English",
+				"libraryCatalog": "Trove",
+				"publisher": "Warradale, S.Aust. : Pioneer Books",
+				"url": "http://trove.nla.gov.au/version/11567057",
+				"attachments": [],
+				"tags": [],
+				"notes": [],
+				"seeAlso": []
+			}
+		]
+	},
+	{
+		"type": "web",
+		"url": "http://trove.nla.gov.au/newspaper/article/70068753",
+		"items": [
+			{
+				"itemType": "newspaperArticle",
+				"title": "'WRAGGE.'",
+				"creators": [],
+				"date": "7 Feb 1903",
+				"abstractNote": "We have received a copy of the above which is a journal devoted chiefly to the science of meteorology. It is owned and conducted by Mr. Clement ...",
+				"libraryCatalog": "Trove",
+				"pages": "4",
+				"place": "Vic.",
+				"publicationTitle": "Sunbury News (Vic. : 1900 - 1910)",
+				"url": "http://nla.gov.au/nla.news-article70068753",
+				"attachments": [
+					{
+						"title": "Trove newspaper PDF",
+						"mimeType": "application/pdf"
+					}
+				],
+				"tags": [
+					"Meteorology Journal - Clement Wragge"
+				],
+				"notes": [
+					{
+						"note": "'WRAGGE' - we have received a copy of the above, which is a journal devoted chiefly to the science of meteorology. It is owned and conducted by Mr. Clement Wragge."
+					}
+				],
+				"seeAlso": []
+			}
+		]
+	},
+	{
+		"type": "web",
+		"url": "http://trove.nla.gov.au/newspaper/result?l-australian=y&q=wragge",
```

Comment:
I get an error testing on this page. The Trove side throws
`main / UI Layout Initialization Warning
The layout-container "main layout / DIV#ui-layout-main" has no height.
Therefore the layout is 0-height and hence 'invisible'!`